packages/react-core/src/components/Select/Select.tsx

```diff
@@ -289,15 +289,17 @@ export class Select extends React.Component<SelectProps & OUIAProps, SelectState
     const { isOpen, isCreatable, createText } = this.props;
     const { typeaheadActiveChild, typeaheadCurrIndex } = this.state;
     if (isOpen) {
-      if (position === 'enter' && (typeaheadActiveChild || this.refCollection[0])) {
-        this.setState({
-          typeaheadInputValue:
-            (typeaheadActiveChild && typeaheadActiveChild.innerText) || this.refCollection[0].innerText
-        });
-        if (typeaheadActiveChild) {
-          typeaheadActiveChild.click();
-        } else {
-          this.refCollection[0].click();
+      if (position === 'enter') {
+        if (typeaheadActiveChild || this.refCollection[0]) {
+          this.setState({
+            typeaheadInputValue:
+              (typeaheadActiveChild && typeaheadActiveChild.innerText) || this.refCollection[0].innerText
+          });
+          if (typeaheadActiveChild) {
+            typeaheadActiveChild.click();
+          } else {
+            this.refCollection[0].click();
+          }
```

Comment:
It's not clear how this fixes the endless loop? Other than splitting the `if` statement to a new line, the structure looks the same to me.

Reply:
Because `this.refCollection[0]` is null in this case the code goes into the else case. I expected that this is not the expected behavior, but as said, I'm not 100% sure here.

The endless loop then happen when `getNextIndex` was called:

https://github.com/patternfly/patternfly-react/blob/b1addef8bca3ad4c4bb851aebb4b5782c45e4ce1/packages/react-core/src/components/Select/Select.tsx#L309

getNextIndex calls getNextIndex again and again:

https://github.com/patternfly/patternfly-react/blob/b1addef8bca3ad4c4bb851aebb4b5782c45e4ce1/packages/react-core/src/helpers/util.ts#L215

Reply:
Thanks for the explanation